src/state/opamEnv.ml

```diff
@@ -658,7 +658,14 @@ let compute_updates ?(force_path=false) st =
     in
     List.map resolve_separator_and_format updates
   in
-  switch_env @ pkg_env @ man_path @ [path]
+  let nix_env =
+    let open OpamFilename in
+    create (OpamPath.Switch.meta OpamStateConfig.(!r.root_dir) st.switch) (basename (raw "nix.env"))
+    |> OpamFile.make
+    |> OpamFile.Environment.read_opt
+    |> Option.fold ~none:[] ~some:(List.map resolve_separator_and_format)
```

Comment:
I'm unsure this is the right place for this code. The function was only dealing with the build environment but this section now reads a file unconditionally (should it be done only when nix is detected?)

Reply:
I think this fails gracefully if the file isn't there with the `Option.fold`, but perhaps we could add a check to only if `os-family = nixos`? I can also refactor this into a separate function `compute_nix_update` called my `OpamEnv.updates`?